client/src/components/Demo/ContractBtns.jsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from "react";
+import useEth from "../../contexts/EthContext/useEth";
+
+function ContractBtns({ setValue }) {
+  const { state: { contract, accounts } } = useEth();
+  const [inputValue, setInputValue] = useState("");
+
+  const handleInputChange = e => {
+    if (/^\d+$|^$/.test(e.target.value)) {
+      setInputValue(e.target.value);
+    }
+  };
+
+  const read = async () => {
+    const value = await contract.methods.read().call({ from: accounts[0] });
+    setValue(value);
+  };
+
+  const write = async e => {
+    if (e.target.tagName === "INPUT") {
+      return;
+    }
+    if (inputValue === "") {
+      alert("Please enter a value to write.");
+      return;
+    }
+    const newValue = parseInt(inputValue);
+    await contract.methods.write(newValue).send({ from: accounts[0] });
+  };
+
+  return (
+    <div className="btns">
+
+      <button onClick={read}>
+        read()
+      </button>
+
+      <div onClick={write} className="input-btn">
+        write(<input
+          type="text"
+          placeholder="uint"
+          value={inputValue}
+          onChange={handleInputChange}
+        />)
+      </div>
+
+    </div>
+  );
+}
+
+export default ContractBtns;
```

Comment:
This is sickkkkkk. My only small concern is that do you think someone who is new to development may be confused by what's going on here? Clearly up to you to decide but just wanted to point out that the boxes are meant to be a bit more of a scaffold. Though these days they're starting to become a lot more complex which I sorta like. Anyway just a thought, no action items on this.